samples/boot/saml2login/src/integration-test/java/org/springframework/security/samples/Saml2LoginIntegrationTests.java

```diff
@@ -123,95 +133,116 @@ public void testRelayState() throws Exception {
-				.andExpect(status().is3xxRedirection())
-				.andExpect(redirectedUrl("/login?error"))
+		sendResponse(response, "/login?error")
 				.andExpect(unauthenticated());
 	}
 
 	@Test
 	public void signedResponseEncryptedAssertion() throws Exception {
-		final String username = "testuser@spring.security.saml";
-		Assertion assertion = buildAssertion(username);
+		Assertion assertion = buildAssertion(USERNAME);
 		EncryptedAssertion encryptedAssertion =
 				OpenSamlActionTestingSupport.encryptAssertion(assertion, decodeCertificate(spCertificate));
 		Response response = buildResponse(encryptedAssertion);
 		signXmlObject(assertion, getSigningCredential(idpCertificate, idpPrivateKey, UsageType.SIGNING));
-		String xml = toXml(response);
-		final ResultActions actions = mockMvc
-				.perform(post("http://localhost:8080/login/saml2/sso/simplesamlphp")
-						.contentType(MediaType.APPLICATION_FORM_URLENCODED)
-						.param("SAMLResponse", OpenSamlActionTestingSupport.encode(xml.getBytes(UTF_8))))
-				.andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/"))
-				.andExpect(authenticated().withUsername(username));
+		sendResponse(response, "/")
+				.andExpect(authenticated().withUsername(USERNAME));
 	}
 
 	@Test
 	public void unsignedResponseEncryptedAssertion() throws Exception {
-		final String username = "testuser@spring.security.saml";
-		Assertion assertion = buildAssertion(username);
+		Assertion assertion = buildAssertion(USERNAME);
 		EncryptedAssertion encryptedAssertion =
 				OpenSamlActionTestingSupport.encryptAssertion(assertion, decodeCertificate(spCertificate));
 		Response response = buildResponse(encryptedAssertion);
-		String xml = toXml(response);
-		final ResultActions actions = mockMvc
-				.perform(post("http://localhost:8080/login/saml2/sso/simplesamlphp")
-						.contentType(MediaType.APPLICATION_FORM_URLENCODED)
-						.param("SAMLResponse", OpenSamlActionTestingSupport.encode(xml.getBytes(UTF_8))))
-				.andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/"))
-				.andExpect(authenticated().withUsername(username));
+		sendResponse(response, "/")
+				.andExpect(authenticated().withUsername(USERNAME));
 	}
 
 	@Test
 	public void signedResponseEncryptedNameId() throws Exception {
-		final String username = "testuser@spring.security.saml";
-		Assertion assertion = buildAssertion(username);
+		Assertion assertion = buildAssertion(USERNAME);
 		final EncryptedID nameId = encryptNameId(assertion.getSubject().getNameID(), decodeCertificate(spCertificate));
 		assertion.getSubject().setEncryptedID(nameId);
 		assertion.getSubject().setNameID(null);
 		Response response = buildResponse(assertion);
 		signXmlObject(assertion, getSigningCredential(idpCertificate, idpPrivateKey, UsageType.SIGNING));
+		sendResponse(response, "/")
+				.andExpect(authenticated().withUsername(USERNAME));
+	}
+
+	@Test
+	public void invalidSignature() throws Exception {
```

Comment:
I think I failed to communicate this in earlier reviews of other PRs, so my apologies for the late advice, but would you please name your tests as:

```java
public void methodUnderTestWhenConditionsThenBehavior()
```

Reply:
If we move to JUnit 5, the `@DisplayName` annotation can make these easy to read sentences.

Reply:
Methods renamed

Reply:
Did these changes already make it into the PR? I'm still seeing the old method names.

Reply:
I missed this test file. Thanks for the double check.